### ISSUE TITLE
In Python environment my below code is oky for Transfer learning or shall i need to add transfer=True?

### ISSUE THREAD

**Issue by TapendraBaduwal** [ultralytics/ultralytics#2351]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

from ultralytics import YOLO

# Load a model
model = YOLO('yolov8n.yaml')  # build a new model from YAML
model = YOLO('yolov8n.pt')  # load a pretrained model (recommended for training)
model = YOLO('yolov8n.yaml').load('yolov8n.pt')  # build from YAML and transfer weights

# Train the model
model.train(data='newdata.yaml', epochs=100, imgsz=640)

**Comment by AbdesselamFerdi** [comment]
@TapendraBaduwal there is no transfer argument in train mode. In short, transfer learning means leveraging the knowledge learned from one task to another task. So, to train the YOLOv8 detector with transfer learning, you need first to initialize its weights with a pre-trained COCO weights (yolov8n.pt) and then train the model on your new dataset.
Hope this help you!

**Comment by TapendraBaduwal** [comment]
@AbdesselamFerdi If i train model with new labels each times with transfer learning concept then how to update and know nc: 80  # number of classes in each time if we train on new labels.....what is the python script to know the number of classes on current yolov8n.pt model

**Comment by AbdesselamFerdi** [comment]
The YOLOv8n detector (yolov8n.pt) has been trained on COCO dataset which contains 80 classes, and you can check this in 'coco.yaml' file in the path 'yolov8\ultralytics\datasets' or in 'yolov8n.yaml' file in the path 'yolov8\ultralytics\models\v8'. 
To train the YOLOv8n detector on your custom dataset, you need to create a 'custom-dataset.yaml' file in the path 'yolov8\ultralytics\datasets' and then define the paths of train/valid/test sets with the class names as in 'coco.yaml' file. After that, change the nc parameter in 'yolov8n.yaml' file to the number of classes of your custom dataset.

**Comment by TapendraBaduwal** [comment]
@AbdesselamFerdi My question is If i train model with  transfer learning concept with new labels name that didi not exist in previous model file each times...i.e lets old nc =80 classes and my new labels class = 10....How to update each time my nc class = 80 +10 = 90 class.....in next time for new labels =10 then latest nc= 90 +10 = 100 like that

**Comment by AbdesselamFerdi** [comment]
One solution is to train the detector multiple times. For example, lets the total number of training epochs is 100 and the number of new classes is 50, then you can divide the new classes into 5 groups thus the number of classes for each group is 10. Next, train the detector 5 times with 20 epochs for each class group (which starts from 80+10 classes, then 90+10 classes and so on) and for each training load the detector with the last trained weights.

**Comment by glenn-jocher** [comment]
@TapendraBaduwal a simple way to update the number of classes for each new training is by changing the 'nc' parameter in the 'yolov8n.yaml' file to the total number of classes for the new training (including the previous classes). For example, if the previous training had 80 classes and the current training has 10 new classes, then you should set 'nc' to 90 in the 'yolov8n.yaml' file. This way, the new training will start with 90 classes, which include the previous 80 classes and the 10 new classes. However, you need to be careful when choosing the 'nc' parameter that the old and new labels should not overlap or else it may cause conflicts.

**Comment by TapendraBaduwal** [comment]
@glenn-jocher How to solve choosing the 'nc' parameter that the old and new labels should not overlap problem ?

Since in my old model have 80 class with each unique id...but my new annotated data have new label with unique id which may match with old label class id... what can i do in this case...how to get class id and label name from pretrain model to map old and new class id?

**Comment by glenn-jocher** [comment]
@TapendraBaduwal, when updating the number of classes for each new training, it is important to avoid overlapping between old and new labels. In case the newly added class already exists in the previous training data, you have to ensure that the new class has a unique ID and label name that does not overlap with any previous class ID or label name. 

If the newly added class overlaps with a previous class ID, it is better to re-annotate the dataset to avoid confusion. However, if the overlap is only in the label name, you can use the 'names' field in the 'custom-dataset.yaml' file (for your newly annotated data) to map the old and new class labels. 
If the old and new training data labels have the same class ID, You can get a list of class IDs and their corresponding labels names from the pre-trained YOLOv8n detector (yolov8n.pt), using the 'coco.names' file in the path 'yolov8/data'. You can then check the new labels against this list to make sure there are no overlapping IDs.
